homeassistant/components/advantage_air/select.py

```diff
@@ -0,0 +1,54 @@
+"""Select platform for Advantage Air integration."""
+
+from homeassistant.components.select import SelectEntity
+
+from .const import DOMAIN as ADVANTAGE_AIR_DOMAIN
+from .entity import AdvantageAirEntity
+
+ADVANTAGE_AIR_INACTIVE = "Inactive"
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up AdvantageAir toggle platform."""
+
+    instance = hass.data[ADVANTAGE_AIR_DOMAIN][config_entry.entry_id]
+
+    entities = []
+    for ac_key in instance["coordinator"].data["aircons"]:
+        entities.append(AdvantageAirMyZone(instance, ac_key))
+    async_add_entities(entities)
+
+
+class AdvantageAirMyZone(AdvantageAirEntity, SelectEntity):
+    """Representation of Advantage Air MyZone control."""
+
+    _attr_icon = "mdi:home-thermometer"
+    _attr_options = [ADVANTAGE_AIR_INACTIVE]
+
+    def __init__(self, instance, ac_key):
+        """Initialize an Advantage Air MyZone control."""
+        super().__init__(instance, ac_key)
+        self._attr_name = f'{self._ac["name"]} MyZone'
+        self._attr_unique_id = (
+            f'{self.coordinator.data["system"]["rid"]}-{ac_key}-myzone'
+        )
+
+        self.number_to_name = {0: ADVANTAGE_AIR_INACTIVE}
+        self.name_to_number = {ADVANTAGE_AIR_INACTIVE: 0}
```

Comment:
These mappings are usually declared outside of the class.

Shouldn't there be more mappings ?

Reply:
If you look at line 41-42, the mappings are built dynamically based on zones present in the data coordinator. Inactive is the neutral "off" state that is always present.

These probably could be put outside the init, but their values are unique to each instance of the class, and thought it was best to leave them near the code block they related to.

Reply:
I moved them to private attributes outside of init